src/Storages/System/StorageSystemReplicas.cpp

```diff
@@ -212,6 +213,7 @@ StorageSystemReplicas::StorageSystemReplicas(const StorageID & table_id_)
         { "can_become_leader",                    std::make_shared<DataTypeUInt8>(),    "Whether the replica can be a leader."},
         { "is_readonly",                          std::make_shared<DataTypeUInt8>(),    "Whether the replica is in read-only mode. This mode is turned on if the config does not have sections with ClickHouse Keeper, "
                                                                                           "if an unknown error occurred when reinitializing sessions in ClickHouse Keeper, and during session reinitialization in ClickHouse Keeper."},
+        { "readonly_duration",                    std::make_shared<DataTypeNullable>(std::make_shared<DataTypeUInt64>()), "Duration that the replica has been readonly for, in milliseconds." },
```

Comment:
I think it would have been much easier just to track the time when a replica became read-only. Then the `readonly_duration` can be obtained via SQL query as `now() - readonly_start_time`

Reply:
Moreover the `readonly_duration` column could be specified as an ALIAS: https://clickhouse.com/docs/en/sql-reference/statements/create/table#alias